operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java

```diff
@@ -24,16 +24,24 @@
 public class Operator implements AutoCloseable, LifecycleAware {
   private static final Logger log = LoggerFactory.getLogger(Operator.class);
   private final KubernetesClient kubernetesClient;
+  private final boolean closeKubernetesClientOnStop;
   private final ConfigurationService configurationService;
   private final ControllerManager controllers = new ControllerManager();
 
+
   public Operator(ConfigurationService configurationService) {
-    this(new DefaultKubernetesClient(), configurationService);
+    this(new DefaultKubernetesClient(), configurationService, true);
```

Comment:
Maybe the default should be `false`, no?

Reply:
Mentioned it in the issues (linked) IMHO usually (at least in our samples) it is used this way, that the registered hook shuts down the operator and also the client.

Reply:
Ok, we need to document it, I guess :)

Reply:
Actually, can we put this configuration option on the `ConfigurationService`? I think it makes sense to put it there and would avoid creating another constructor.

Reply:
good point, will change it a look how it would look like

Reply:
updated